### PR TITLE
Switch package management to go modules

### PR DESCRIPTION
**Pull request by austinabell**
Previously used dep, but go modules is preferred since it removes the requirement of having to run code from the $GOPATH among other improvements.

Still uses Alexander's fork of go-ethereum with:

```
replace github.com/ethereum/go-ethereum v1.8.27 => github.com/alexanderbez/go-ethereum v1.8.17-0.20181024144731-0a57b29f0c8e
```

But we can research and discus how and what to replace this dependency with.

I didn't thoroughly check which makefile commands are nullified by this change so if you see something please let me know!

Also should looking into updating the cosmos sdk version, to update to their updated structure and functionality since the commit that is used as dependency (v0.28.0).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,32 +19,37 @@ DOCKER_TAG = unstable
 DOCKER_IMAGE = cosmos/ethermint
 ETHERMINT_DAEMON_BINARY = emintd
 ETHERMINT_CLI_BINARY = emintcli
+GO_MOD=GO111MODULE=on
```
Comment (ansermino) [inline Makefile:22]: It should be possible to simplify this to `GO111MODULE=on`, with the additions below removed. Once its declared here it is within the scope of all the commands this file calls (eg. `go build`).

**Review by ansermino — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (austinabell) [reply Makefile:22]: Tried this, doesn't work when GO111MODULE=on is not exported and code is run within the $GOPATH. I ran into this issue before and was very picky about being included before every go command which required the dependencies.

edit: here is the output:
```
~/go/src/github.com/cosmos/ethermint (austin/modules) $  make test-import
# github.com/cosmos/ethermint/importer
package github.com/cosmos/ethermint/importer (test)
	imports github.com/cosmos/ethermint/x/evm/types
	imports github.com/ethereum/go-ethereum/crypto/sha3: cannot find package "github.com/ethereum/go-ethereum/crypto/sha3" in any of:
	/usr/local/Cellar/go/1.12.1/libexec/src/github.com/ethereum/go-ethereum/crypto/sha3 (from $GOROOT)
	/Users/austinabell/go/src/github.com/ethereum/go-ethereum/crypto/sha3 (from $GOPATH)
FAIL	github.com/cosmos/ethermint/importer [setup failed]
make: *** [test-import] Error 1
```

Comment (ansermino) [reply Makefile:22]: It seems to work for me if I just make this change:
```
GO111MODULE=on
GO_MOD=
```
Any chance you have `GO111MODULE` set on your machine already? Maybe in `.bashrc` or `.profile`?

**Review by GregTheGreek — APPROVED** [review]
~Everything~ `make install` & `make build` work on my machine with the following settings:
```
➜  ethermint git:(austin/modules) go version
go version go1.12.5 darwin/amd64
➜  ethermint git:(austin/modules) echo $GO111MODULE
on
```

File: Makefile (hunk 1 above)
Comment (austinabell) [reply Makefile:22]: Doesn't work to just do this when run in $GOPATH, although is sufficient for running code outside of $GOPATH.

**Review by ansermino — APPROVED** [review]

**Pull request merged**